lib/project_types/script/layers/infrastructure/assemblyscript_project_creator.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class AssemblyScriptProjectCreator
+        BOOTSTRAP_SRC = "npx --no-install shopify-scripts-bootstrap src %{src_base}"
+        BOOTSTRAP_TEST = "npx --no-install shopify-scripts-bootstrap test %{test_base}"
+        SOURCE_DIR = "src"
+        TEST_DIR = "test"
+        LANGUAGE = "ts"
+
+        attr_reader :ctx
+
+        def initialize(ctx, extension_point, script_name)
+          @ctx = ctx
+          @extension_point = extension_point
+          @script_name = script_name
+        end
+
+        def setup_dependencies
+          write_npmrc
+          write_package_json
+        end
+
+        def bootstrap
+          create_src_folder
+          create_test_folder
+        end
+
+        private
+
+        def project
+          @project ||= ScriptProject.current
+        end
+
+        def create_src_folder
+          FileUtils.mkdir_p(src_base)
```

Comment:
Note that I’ll refactor these to use `ctx` once https://github.com/Shopify/shopify-app-cli/pull/689 merges. I’m missing some of his new methods so I’ll just wait to do them all